src/Tracing/SamplingContext.php

```diff
@@ -48,4 +53,24 @@ public function setParentSampled(?bool $parentSampled): void
     {
         $this->parentSampled = $parentSampled;
     }
+
+    /**
+     * Sets additional data that will be provided as a second argument to {@link \Sentry\startTransaction()}.
+     *
+     * @param array<string, mixed>|null $additionalContext
+     */
+    public function setAdditionalContext(?array $additionalContext): void
+    {
+        $this->additionalContext = $additionalContext;
+    }
```

Comment:
IDK how much this difference matters, but in the other SDKs, the context data provided in `customSamplingContext` is spread into `samplingContext`, whereas here it seems like it's being added as a single separate entry called `additionalContext`.

Reply:
I would say this is ok to be not 100% the same, spreading this in PHP isn't something that feels natural in PHP.

Reply:
My two cents: it's not really important whether we want to use the spreading operator or not, the key question is if this method should merge the new data with the existing one or not

Reply:
That (merging) is how it works elsewhere. I feel like it's just a smidge nicer not to have to go digging any deeper than you need to for stuff, so I'd be in favor of the spread, but I don't feel strongly.

Reply:
>spreading this in PHP isn't something that feels natural in PHP

Reading again @HazAT comment I think that what he meant is that in PHP a setter sets data, does not merge it, at least usually. I agree with this, but I don't have a strong opinion so I will let you two choose what to do. As far as I'm concerned, PR looks good and I'm waiting this discussion to reach an agreement before giving the final approval

Reply:
>  in PHP a setter sets data, does not merge it

Ah, I see. Thanks for the explanation. Any reason we couldn't do away with this method entirely and just do

```PHP
$samplingContext = array_merge(SamplingContext::getDefault($context), $customSamplingContext));
```

in `startTransaction()`?

Reply:
We cannot because `SamplingContext::getDefault()` return an object instance, so `array_merge` cannot work because you are trying to merge it with an array. Either we don't bother about the setter merging data (it's not a rule written in the stone that a setter must overwrite data rather than merging it) or we could rename the method to something else, e.g. `mergeAdditionalContext`

Reply:
Okay. I poked at it a bit and I guess the real answer here would be to have `SamplingContext` just be a hashmap rather than a full-blown object instance, thereby allowing for arbitrary keys, but I assume that would be a totally breaking change, so... I think it's fine. Let's just leave it.

Thanks for the explanations - it's helpful to know a tiny bit about how PHP works.